interop-cats-effect/shared/src/main/scala/scalaz/zio/interop/bio/Concurrent2.scala

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio
+package interop
+package bio
+
+import cats.kernel.Monoid
+import cats.syntax.option._
+import cats.syntax.flatMap.catsSyntaxFlatten
+import scalaz.zio.interop.bio.data.{ Deferred2, Ref2 }
+
+import scala.concurrent.ExecutionContext
+
+abstract class Concurrent2[F[+ _, + _]] extends Temporal2[F] { self =>
+
+  /**
+   * Returns an effect that runs `fa` into its own separate `Fiber2`
+   * and returns the fiver as result.
+   *
+   * TODO: Example:
+   * {{{
+   *
+   * }}}
+   *
+   */
+  def start[E, A](fa: F[E, A]): F[Nothing, Fiber2[F, E, A]]
+
+  /**
+   * Performs `fa` uninterruptedly. This will prevent it from
+   * being terminated externally, but it may still fail for internal
+   * reasons.
+   *
+   * TODO: Example:
+   * {{{
+   *
+   * }}}
+   *
+   */
+  def uninterruptible[E, A](fa: F[E, A]): F[E, A]
+
+  /**
+   * Executes the `cleanup` effect if `fa` is interrupted
+   *
+   * TODO: Example:
+   * {{{
+   *
+   * }}}
+   *
+   */
+  def onInterrupt[E, A](fa: F[E, A])(cleanup: F[Nothing, _]): F[E, A]
```

Comment:
We can actually implement this one, we just need `bracket`.

Reply:
changed.